client/my-sites/plans/jetpack-plans/product-store/more-info-link/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { EXTERNAL_PRODUCTS_LIST } from '../../constants';
+import { MoreInfoLinkProps } from '../types';
+
+import './style.scss';
+
+export const MoreInfoLink: React.FC< MoreInfoLinkProps > = ( { item, onClickMore } ) => {
+	const translate = useTranslate();
+
+	const isExternalProduct = useMemo(
+		() => EXTERNAL_PRODUCTS_LIST.includes( item.productSlug ),
+		[ item ]
+	);
```

Comment:
This doesn't need memoization IMHO. Direct call to `.includes` will be more than 100 times faster than `useMemo` here.

```suggestion
	const isExternalProduct = EXTERNAL_PRODUCTS_LIST.includes( item.productSlug );
```

Reply:
now that you mentioned this. I'm kinda confuse when exactly are we gonna use memoization. For the following, maybe we shouldn't also useCallback here. 

```
const clearCurrentItem = useCallback( () => {
		setCurrentItem( null );
		window.location.hash = '';
	}, [] );

	const onClickMoreInfoFactory = useCallback( ( item: SelectorProduct ): VoidFunction => {
		return () => {
			recordTracksEvent( 'calypso_product_more_about_product_click', {
				product: item.productSlug,
			} );

			if ( ! EXTERNAL_PRODUCTS_LIST.includes( item.productSlug ) ) {
				setCurrentItem( item );
			}
		};
	}, [] );
```

Reply:
Wherever the computation or logic is expensive or it changes on every render, we should use `useMemo`/`useCallback`.

For example, when you create an object `{}` without `useMemo`, it will be re-created on every render and the reference will change every time. The same goes for functions. If you create a function without `useCallback` it will be a new function (by reference) on every render.
React uses [`Object.is`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) check to decide whether to re-render or not, which uses referential equality check which fails for non-primitive values like objects/functions. `useMemo` and `useCallback` maintain that referential equality unless the dependencies change.

This is not needed for [primitive values](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) (like `isExternalProduct` here) because `Object.is` will check for their values instead of reference. So, `isExternalProduct` will be same for React with our without `useMemo`.

Primitive results should not be memoized unless their compilation is expensive which is not the case with `EXTERNAL_PRODUCTS_LIST.includes()` here.

Hope that clears the confusion.

Reply:
Ahh I get now what you mean.. thank you for clearing this up. It's a good learning for me. 👍